connect/eaas/runner/managers/transformation.py

```diff
@@ -0,0 +1,354 @@
+                f'result: {result.status}, '
+                f'took: {result_message.output.runtime}',
+            )
+            if result.status in (ResultType.SKIP, ResultType.FAIL):
+                result_message.output.message = result.output
+        except Exception as e:
+            self.log_exception(task_data, e)
+            result_message.output = TaskOutput(result=ResultType.RETRY)
+            result_message.output.message = traceback.format_exc()[:4000]
+
+        return result_message
+
+    async def process_transformation(self, task_data, tfn_request, method):
+        input_file = await asyncio.get_running_loop().run_in_executor(
+            self.executor,
+            self.download_excel,
+            tfn_request,
+            task_data.options.api_key,
+        )
+        output_file = NamedTemporaryFile(
+            suffix=f'.{tfn_request["files"]["input"]["name"].split(".")[-1]}',
+        )
+
+        read_queue = asyncio.Queue(TRANSFORMATION_TASK_MAX_PARALLEL_LINES)
+        write_queue = asyncio.Queue()
+
+        loop = asyncio.get_event_loop()
+
+        reader_task = loop.run_in_executor(
+            self.executor,
+            self.read_excel,
+            input_file,
+            read_queue,
+            loop,
+        )
+        writer_task = loop.run_in_executor(
+            self.executor,
+            self.write_excel,
+            output_file.name,
+            write_queue,
+            tfn_request['stats']['rows']['total'],
+            tfn_request['transformation']['columns']['output'],
+            task_data,
+            loop,
+        )
+        processor_task = asyncio.create_task(self.process_rows(
+            read_queue,
+            write_queue,
+            method,
+            tfn_request['stats']['rows']['total'],
+        ))
+
+        tasks = [reader_task, writer_task, processor_task]
+        try:
+            await asyncio.gather(*tasks)
+        except Exception as e:
+            logger.error(
+                f'Error during processing transformation '
+                f'for {task_data.options.task_id}: {e}',
+            )
+            for task in tasks:
+                if not task.done():
+                    task.cancel()
+            input_file.close()
+            output_file.close()
+            client = self.get_client(task_data)
+            await client('billing').requests[task_data.input.object_id]('fail').post()
+            return TransformationResponse.fail(output=str(e))
+
+        await self.send_output_file(task_data, tfn_request['batch']['id'], output_file)
+        input_file.close()
+        output_file.close()
+        return TransformationResponse.done()
+
+    def download_excel(self, tfn_request, api_key):
+        input_file_name = tfn_request['files']['input']['name']
+        input_file = NamedTemporaryFile(suffix=f'.{input_file_name.split(".")[-1]}')
+
+        with requests.get(
+            url=f'{self.config.get_api_url()}{input_file_name}',
```

Comment:
Here you have to expose a property in config "api_address" and use it.

Because get_api_url() returns https://hostname/public/v1 and input_file_name starts with /public/v1.